### PR TITLE
Use modern thread name attribute API instead of the deprecated methods

### LINKED ISSUE #369: DeprecationWarning: currentThread() is deprecated, use current_thread() instead
❓ **I'm submitting a ...**
- [X] 🐞 bug report
- [ ] 🐣 feature request
- [ ] ❓ question about the decisions made in the repository

🐞 **Describe the bug. What is the current behavior?**

When testing on Python 3.10 a DeprecationWarning is emitted from the use of an old alias for `threading.current_thread()` in the following function:

https://github.com/cherrypy/cheroot/blob/670a5c2ef69fb158cbf7ffbf2726818466b55db8/cheroot/workers/threadpool.py#L316-L325

While debugging this I also found several uses of the deprecated "setName" and "getName" in the same file.

❓ **What is the motivation / use case for changing the behavior?**

Uses of deprecated methods should be remedied in advance of the methods being removed.

💡 **To Reproduce**

1. Run Python 3.10 with the ``-Walways`` argument.
2. Run the following commands:

```python
>>> import cherrypy
>>> cherrypy.server.start()
>>> cherrypy.server.stop()
```

3. See lots of warnings

💡 **Expected behavior**

No deprecation warnings were expected.

📋 **Details**
<details>

```python
/usr/lib/python3.10/site-packages/cherrypy/process/servers.py:181: DeprecationWarning: getName() is deprecated, get the name attribute instead
  t.setName('HTTPServer ' + t.getName())
/usr/lib/python3.10/site-packages/cherrypy/process/servers.py:181: DeprecationWarning: setName() is deprecated, set the name attribute instead
  t.setName('HTTPServer ' + t.getName())
/usr/lib/python3.10/site-packages/cheroot/workers/threadpool.py:181: DeprecationWarning: getName() is deprecated, get the name attribute instead
  format(worker_name=worker.getName()),
/usr/lib/python3.10/site-packages/cheroot/workers/threadpool.py:179: DeprecationWarning: setName() is deprecated, set the name attribute instead
  worker.setName(
/usr/lib/python3.10/site-packages/cheroot/workers/threadpool.py:111: DeprecationWarning: getName() is deprecated, get the name attribute instead
  self.server.stats['Worker Threads'][self.getName()] = self.stats
/usr/lib/python3.10/site-packages/cheroot/workers/threadpool.py:181: DeprecationWarning: getName() is deprecated, get the name attribute instead
  format(worker_name=worker.getName()),
/usr/lib/python3.10/site-packages/cheroot/workers/threadpool.py:179: DeprecationWarning: setName() is deprecated, set the name attribute instead
  worker.setName(
/usr/lib/python3.10/site-packages/cheroot/workers/threadpool.py:111: DeprecationWarning: getName() is deprecated, get the name attribute instead
  self.server.stats['Worker Threads'][self.getName()] = self.stats
/usr/lib/python3.10/site-packages/cheroot/workers/threadpool.py:181: DeprecationWarning: getName() is deprecated, get the name attribute instead
  format(worker_name=worker.getName()),
/usr/lib/python3.10/site-packages/cheroot/workers/threadpool.py:179: DeprecationWarning: setName() is deprecated, set the name attribute instead
  worker.setName(
/usr/lib/python3.10/site-packages/cheroot/workers/threadpool.py:111: DeprecationWarning: getName() is deprecated, get the name attribute instead
  self.server.stats['Worker Threads'][self.getName()] = self.stats
/usr/lib/python3.10/site-packages/cheroot/workers/threadpool.py:181: DeprecationWarning: getName() is deprecated, get the name attribute instead
  format(worker_name=worker.getName()),
/usr/lib/python3.10/site-packages/cheroot/workers/threadpool.py:179: DeprecationWarning: setName() is deprecated, set the name attribute instead
  worker.setName(
/usr/lib/python3.10/site-packages/cheroot/workers/threadpool.py:111: DeprecationWarning: getName() is deprecated, get the name attribute instead
  self.server.stats['Worker Threads'][self.getName()] = self.stats
/usr/lib/python3.10/site-packages/cheroot/workers/threadpool.py:181: DeprecationWarning: getName() is deprecated, get the name attribute instead
  format(worker_name=worker.getName()),
/usr/lib/python3.10/site-packages/cheroot/workers/threadpool.py:179: DeprecationWarning: setName() is deprecated, set the name attribute instead
  worker.setName(
/usr/lib/python3.10/site-packages/cheroot/workers/threadpool.py:111: DeprecationWarning: getName() is deprecated, get the name attribute instead
  self.server.stats['Worker Threads'][self.getName()] = self.stats
/usr/lib/python3.10/site-packages/cheroot/workers/threadpool.py:181: DeprecationWarning: getName() is deprecated, get the name attribute instead
  format(worker_name=worker.getName()),
/usr/lib/python3.10/site-packages/cheroot/workers/threadpool.py:179: DeprecationWarning: setName() is deprecated, set the name attribute instead
  worker.setName(
/usr/lib/python3.10/site-packages/cheroot/workers/threadpool.py:111: DeprecationWarning: getName() is deprecated, get the name attribute instead
  self.server.stats['Worker Threads'][self.getName()] = self.stats
/usr/lib/python3.10/site-packages/cheroot/workers/threadpool.py:181: DeprecationWarning: getName() is deprecated, get the name attribute instead
  format(worker_name=worker.getName()),
/usr/lib/python3.10/site-packages/cheroot/workers/threadpool.py:179: DeprecationWarning: setName() is deprecated, set the name attribute instead
  worker.setName(
/usr/lib/python3.10/site-packages/cheroot/workers/threadpool.py:111: DeprecationWarning: getName() is deprecated, get the name attribute instead
  self.server.stats['Worker Threads'][self.getName()] = self.stats
/usr/lib/python3.10/site-packages/cheroot/workers/threadpool.py:181: DeprecationWarning: getName() is deprecated, get the name attribute instead
  format(worker_name=worker.getName()),
/usr/lib/python3.10/site-packages/cheroot/workers/threadpool.py:179: DeprecationWarning: setName() is deprecated, set the name attribute instead
  worker.setName(
/usr/lib/python3.10/site-packages/cheroot/workers/threadpool.py:111: DeprecationWarning: getName() is deprecated, get the name attribute instead
  self.server.stats['Worker Threads'][self.getName()] = self.stats
/usr/lib/python3.10/site-packages/cheroot/workers/threadpool.py:181: DeprecationWarning: getName() is deprecated, get the name attribute instead
  format(worker_name=worker.getName()),
/usr/lib/python3.10/site-packages/cheroot/workers/threadpool.py:179: DeprecationWarning: setName() is deprecated, set the name attribute instead
  worker.setName(
/usr/lib/python3.10/site-packages/cheroot/workers/threadpool.py:111: DeprecationWarning: getName() is deprecated, get the name attribute instead
  self.server.stats['Worker Threads'][self.getName()] = self.stats
/usr/lib/python3.10/site-packages/cheroot/workers/threadpool.py:181: DeprecationWarning: getName() is deprecated, get the name attribute instead
  format(worker_name=worker.getName()),
/usr/lib/python3.10/site-packages/cheroot/workers/threadpool.py:179: DeprecationWarning: setName() is deprecated, set the name attribute instead
  worker.setName(
/usr/lib/python3.10/site-packages/cheroot/workers/threadpool.py:111: DeprecationWarning: getName() is deprecated, get the name attribute instead
  self.server.stats['Worker Threads'][self.getName()] = self.stats
[16/May/2021:07:50:46] ENGINE Serving on http://127.0.0.1:8080
/usr/lib/python3.10/site-packages/cheroot/workers/threadpool.py:323: DeprecationWarning: currentThread() is deprecated, use current_thread() instead
  if thread is not threading.currentThread()
/usr/lib/python3.10/site-packages/cheroot/workers/threadpool.py:323: DeprecationWarning: currentThread() is deprecated, use current_thread() instead
  if thread is not threading.currentThread()
/usr/lib/python3.10/site-packages/cheroot/workers/threadpool.py:323: DeprecationWarning: currentThread() is deprecated, use current_thread() instead
  if thread is not threading.currentThread()
/usr/lib/python3.10/site-packages/cheroot/workers/threadpool.py:323: DeprecationWarning: currentThread() is deprecated, use current_thread() instead
  if thread is not threading.currentThread()
/usr/lib/python3.10/site-packages/cheroot/workers/threadpool.py:323: DeprecationWarning: currentThread() is deprecated, use current_thread() instead
  if thread is not threading.currentThread()
/usr/lib/python3.10/site-packages/cheroot/workers/threadpool.py:323: DeprecationWarning: currentThread() is deprecated, use current_thread() instead
  if thread is not threading.currentThread()
/usr/lib/python3.10/site-packages/cheroot/workers/threadpool.py:323: DeprecationWarning: currentThread() is deprecated, use current_thread() instead
  if thread is not threading.currentThread()
/usr/lib/python3.10/site-packages/cheroot/workers/threadpool.py:323: DeprecationWarning: currentThread() is deprecated, use current_thread() instead
  if thread is not threading.currentThread()
/usr/lib/python3.10/site-packages/cheroot/workers/threadpool.py:323: DeprecationWarning: currentThread() is deprecated, use current_thread() instead
  if thread is not threading.currentThread()
/usr/lib/python3.10/site-packages/cheroot/workers/threadpool.py:323: DeprecationWarning: currentThread() is deprecated, use current_thread() instead
  if thread is not threading.currentThread()
[16/May/2021:07:50:46] ENGINE HTTP Server cherrypy._cpwsgi_server.CPWSGIServer(('127.0.0.1', 8080)) shut down

```

</details>

📋 **Environment**

- Cheroot version: 8.5.2
- CherryPy version: 18.6.0
- Python version: 3.10.0b1
- OS: Ubuntu 20.04
- Browser: N/A

📋 **Additional context**

Looking at the [Python documentation](https://docs.python.org/2.7/library/threading.html) the new names were added in Python 2.6, so they should be available on all versions you support.

### PR DESCRIPTION
**Pull request by soham4abc**
❓ **What kind of change does this PR introduce?**
  - [x] 🐞 bug fix
  - [ ] 🐣 feature
  - [ ] 📋 docs update
  - [ ] 📋 tests/coverage improvement
  - [ ] 📋 refactoring
  - [ ] 💥 other

📋 **What is the related issue number (starting with `#`)**

Resolves #369 

📋 **Checklist**:

  - [x] I think the code is well written
  - [x] I wrote [good commit messages][1] 
  - [x] The PR relates to *only* one subject with a clear title
  and description in grammatically correct, complete sentences

[1]: http://chris.beams.io/posts/git-commit/
[2]: https://github.com/todotxt/todo.txt-android/wiki/Squash-All-Commits-Related-to-a-Single-Issue-into-a-Single-Commit

---
This change is [Reviewable](https://reviewable.io/reviews/cherrypy/cheroot/371)

### CODE REVIEW

**Comment by soham4abc** [comment]
my bad

**Comment by webknjaz** [comment]
Do you want to try again? The diff looks fine

**Comment by soham4abc** [comment]
> Do you want to try again? The diff looks fine 

The builds failed miserably. If you want I can re issue the PR.

**Comment by soham4abc** [comment]
looks good?

**Comment by soham4abc** [comment]
Mail says PR run failed

**Comment by webknjaz** [comment]
The docs jobs failures are unrelated and the test matrix didn't get fully executed — maybe GHA was having some internal platform issues. FWIW I restarted the test jobs to see how it goes.

File: cheroot/workers/threadpool.py [hunk 1]
```diff
@@ -108,7 +108,7 @@ def run(self):
 
         Retrieves incoming connections from thread pool.
         """
-        self.server.stats['Worker Threads'][self.getName()] = self.stats
+        self.server.stats['Worker Threads'][self.get_name()] = self.stats
```
Comment (webknjaz) [inline cheroot/workers/threadpool.py:111]: I'm pretty sure it's
```suggestion
        self.server.stats['Worker Threads'][self.name] = self.stats
```

File: cheroot/workers/threadpool.py [hunk 2]
```diff
@@ -176,9 +176,9 @@ def start(self):
         for i in range(self.min):
             self._threads.append(WorkerThread(self.server))
         for worker in self._threads:
-            worker.setName(
+            worker.set_name(
                 'CP Server {worker_name!s}'.
-                format(worker_name=worker.getName()),
+                format(worker_name=worker.get_name()),
             )
```
Comment (webknjaz) [inline cheroot/workers/threadpool.py:182]:
```suggestion
            worker.name = (
                'CP Server {worker_name!s}'.
                format(worker_name=worker.name),
            )
```

File: cheroot/workers/threadpool.py [hunk 3]
```diff
@@ -226,9 +226,9 @@ def grow(self, amount):
 
     def _spawn_worker(self):
         worker = WorkerThread(self.server)
-        worker.setName(
+        worker.set_name(
```
Comment (webknjaz) [inline cheroot/workers/threadpool.py:229]:
```suggestion
        worker.name = (
```

File: cheroot/workers/threadpool.py [hunk 4]
```diff
@@ -226,9 +226,9 @@ def grow(self, amount):
...
             'CP Server {worker_name!s}'.
-            format(worker_name=worker.getName()),
+            format(worker_name=worker.get_name()),
```
Comment (webknjaz) [inline cheroot/workers/threadpool.py:231]:
```suggestion
            format(worker_name=worker.name),
```

File: cheroot/workers/threadpool.py [hunk 5]
```diff
@@ -176,9 +176,9 @@ def start(self):
         for i in range(self.min):
             self._threads.append(WorkerThread(self.server))
         for worker in self._threads:
-            worker.setName(
+            worker.set_name(
```
Comment (webknjaz) [inline cheroot/workers/threadpool.py:179]:
```suggestion
            worker.name = (
```

File: cheroot/workers/threadpool.py [hunk 6]
```diff
@@ -176,9 +176,9 @@ def start(self):
...
                 'CP Server {worker_name!s}'.
-                format(worker_name=worker.getName()),
+                format(worker_name=worker.get_name()),
```
Comment (webknjaz) [inline cheroot/workers/threadpool.py:181]:
```suggestion
                format(worker_name=worker.name),
```

**Pull request merged**